sdk/src/Core/Amazon.Runtime/Internal/Auth/ISigner.cs

```diff
@@ -24,11 +31,25 @@ namespace Amazon.Runtime.Internal.Auth
     /// the service to authenticate the SDK customer's identity.
     /// </para>
     /// </summary>
-#pragma warning disable CA1040 // Avoid empty interfaces
     public interface ISigner
```

Comment:
Should add /// comments to the other public methods since you added them to RequiresCredentials.

Reply:
@muhammad-othman There's a blob in the SRA book if you want to use that: _Signs the provided HTTP request. The provided identity is used for signing, and an exception will be thrown if the identity is not compatible with this signer._

I also noticed the existing comment for `RequiresCredentials` mentions immutable credentials (which we're trying to remove). May as well remove the `<see cref="ImmutableCredentials"/>` from there.